src/asmcnc/core_UI/popups.py

```diff
@@ -0,0 +1,280 @@
+        self,
+        main_string,
+        popup_type,
+        main_label_padding,
+        main_layout_padding,
+        main_layout_spacing,
+        main_label_size_delta,
+        button_layout_padding,
+        button_layout_spacing,
+        popup_width,
+        popup_height,
+        popup_image=None,
+        popup_image_size_hint=None,
+        button_one_text="Ok",
+        button_one_callback=None,
+        button_one_background_color=None,
+        button_two_text=None,
+        button_two_callback=None,
+        button_two_background_color=None,
+        **kwargs
+    ):
+        super(BasicPopup, self).__init__(**kwargs)
+
+        if button_one_callback is None:
+            button_one_callback = self.dismiss
+        if button_one_background_color is not None:
+            self.button_one_background_normal = ""
+        if button_two_background_color is not None:
+            self.button_two_background_down = ""
+
+        self.title = self.l.get_str(kwargs["title"])
+        self.size_hint = (None, None)
+        self.width = dp(utils.get_scaled_width(popup_width))
+        self.height = dp(utils.get_scaled_height(popup_height))
+        self.popup_width = popup_width
+        self.popup_height = popup_height
+        self.main_string = self.l.get_str(main_string)
+        self.main_label_padding = main_label_padding
+        self.main_layout_padding = main_layout_padding
+        self.main_label_size_delta = main_label_size_delta
+        self.button_layout_padding = button_layout_padding
+        self.button_layout_spacing = button_layout_spacing
+        self.main_layout_spacing = main_layout_spacing
+        self.popup_type = popup_type
+        self.popup_image = popup_image
+        self.popup_image_size_hint = popup_image_size_hint
+        self.button_one_text = self.l.get_str(button_one_text)
+        self.button_one_callback = button_one_callback
+        self.button_two_text = (
+            self.l.get_str(button_two_text) if button_two_text is not None else None
+        )
+        self.button_two_callback = button_two_callback
+        self.button_one_background_color = button_one_background_color
+        self.button_two_background_color = button_two_background_color
+        self.button_one_background_normal = (
+            self.button_one_background_normal
+            if button_one_background_color is None
+            else ""
+        )
+        self.button_two_background_normal = (
+            self.button_two_background_normal
+            if button_two_background_color is None
+            else ""
+        )
+        if self.popup_type is not None:
+            if self.popup_type.value is not None:
+                self.separator_color = self.popup_type.value["separator_color"]
+
+        self.build()
+
+    def build(self):
+        text_size_x = dp(
+            utils.get_scaled_width(
+                self.popup_width - utils.get_scaled_width(self.main_label_size_delta)
+            )
```

Comment:
I'm confused, this looks like it shouldn't work because I read it as: 

`scale(unscaled_width - scaled_delta)` 

??